l1-contracts/src/core/Rollup.sol

```diff
@@ -156,6 +160,59 @@ contract Rollup is Leonidas, IRollup, ITestRollup {
     vkTreeRoot = _vkTreeRoot;
   }
 
+  function claimEpochProofRight(DataStructures.EpochProofQuote calldata _quote)
+    external
+    override(IRollup)
+  {
+    uint256 currentSlot = getCurrentSlot();
+    address currentProposer = getCurrentProposer();
+    uint256 epochToProve = getEpochToProve();
+
+    if (currentProposer != address(0) && currentProposer != msg.sender) {
+      revert Errors.Leonidas__InvalidProposer(currentProposer, msg.sender);
+    }
+
+    if (_quote.epochToProve != epochToProve) {
+      revert Errors.Rollup__NotClaimingCorrectEpoch(epochToProve, _quote.epochToProve);
+    }
+
+    if (currentSlot % Constants.AZTEC_EPOCH_DURATION >= CLAIM_DURATION_IN_L2_SLOTS) {
+      revert Errors.Rollup__NotInClaimPhase(
+        currentSlot, currentSlot % Constants.AZTEC_EPOCH_DURATION
+      );
+    }
+
+    // Overwrite stale proof claims:
+    // if the epoch to prove is not the one that has been claimed,
+    // then whatever is in the proofClaim is stale,
+    // and we overwrite below
+    if (proofClaim.epochToProve == epochToProve && proofClaim.proposerClaimant != address(0)) {
+      revert Errors.Rollup__ProofRightAlreadyClaimed();
+    }
+
+    if (_quote.bondAmount < PROOF_COMMITMENT_MIN_BOND_AMOUNT_IN_TST) {
+      revert Errors.Rollup__InsufficientBondAmount(
+        PROOF_COMMITMENT_MIN_BOND_AMOUNT_IN_TST, _quote.bondAmount
+      );
+    }
+
+    if (_quote.validUntilSlot < currentSlot) {
+      revert Errors.Rollup__QuoteExpired(currentSlot, _quote.validUntilSlot);
+    }
+
+    address bondProvider = PROOF_COMMITMENT_ESCROW.stakeBond(_quote);
```

Comment:
You stake, but there won't ever be an unstake. Bye bye money. See comment in other pr.

Reply:
I was waiting to unstake the bond until the proof of epoch landed.

I filed https://github.com/AztecProtocol/aztec-packages/issues/8652.